tests/testthat/test-teal_slice-store.R

```diff
@@ -128,3 +128,46 @@ test_that("teal_slice store/restore restores mixed `Date`-characters as characte
   tss_restored <- slices_restore(slices_path)
   expect_identical_slice(tss[[1]], tss_restored[[1]])
 })
+
+
+
+test_that(
+  "teal_slice store/restore restores factors as characters in selected and choices because they are converted",
```

Comment:
I think this is better tested on `teal_slice`, that's where the conversion takes place.